source/common/logger/ur_logger_details.hpp

```diff
@@ -59,8 +64,43 @@ class Logger {
         }
     }
 
+    template <typename... Args>
+    void debug(const logger::LegacyMessage &p, const char *format,
+               Args &&...args) {
+        log(p, logger::Level::DEBUG, format, std::forward<Args>(args)...);
+    }
+
+    template <typename... Args>
+    void info(const logger::LegacyMessage &p, const char *format,
+              Args &&...args) {
+        log(p, logger::Level::INFO, format, std::forward<Args>(args)...);
+    }
+
+    template <typename... Args>
+    void warning(const logger::LegacyMessage &p, const char *format,
+                 Args &&...args) {
+        log(p, logger::Level::WARN, format, std::forward<Args>(args)...);
+    }
+
+    template <typename... Args>
+    void error(const logger::LegacyMessage &p, const char *format,
+               Args &&...args) {
+        log(p, logger::Level::ERR, format, std::forward<Args>(args)...);
+    }
+
     template <typename... Args>
     void log(logger::Level level, const char *format, Args &&...args) {
+        log(logger::LegacyMessage(format), level, format,
+            std::forward<Args>(args)...);
+    }
+
+    template <typename... Args>
+    void log(const logger::LegacyMessage &p, logger::Level level,
+             const char *format, Args &&...args) {
+        if (isLegacySink) {
+            sink->log(level, p.message, std::forward<Args>(args)...);
```

Comment:
Shouldn't you check if the sink is not a nullptr first?

Reply:
Done